src/claimant.js

```diff
@@ -0,0 +1,171 @@
+import xdr from './generated/stellar-xdr_generated';
+import { Keypair } from './keypair';
+import { StrKey } from './strkey';
+
+/**
+ * Claimant class represents an xdr.Claimant
+ *
+ * The claim predicate is optional, it defaults to unconditional if none is specified.
+ *
+ * @constructor
+ * @param {string} destination - The destination account ID.
+ * @param {xdr.ClaimPredicate} [predicate] - The claim predicate.
+ */
+export class Claimant {
+  constructor(destination, predicate) {
+    if (destination && !StrKey.isValidEd25519PublicKey(destination)) {
+      throw new Error('Destination is invalid');
+    }
+    this._destination = destination;
+
+    if (!predicate) {
+      this._predicate = xdr.ClaimPredicate.claimPredicateUnconditional();
+    } else if (predicate instanceof xdr.ClaimPredicate) {
+      this._predicate = predicate;
+    } else {
+      throw new Error('Predicate should be an xdr.ClaimPredicate');
+    }
+  }
+
+  /**
+   * Returns an unconditional claim predicate
+   * @Return {xdr.ClaimPredicate}
+   */
+  static claimPredicateUnconditional() {
```

Comment:
Pretty nitpicky, but maybe `predicateUnconditional` is enough here (and in the others), given the `Claimant` prefix? Could reduce code verbosity, e.g.
```js
Claimant.predicateUnconditional()
// vs.
Claimant.claimPredicateUnconditional()
```
especially as they get combined. No strong opinion here, just a thought.

Reply:
good idea! I've updated the function names.